### PR TITLE
Ports Bubblegum

### PR DESCRIPTION
**Pull request by LadyFowls**
Ports bubblegum

- [x ] Get the code ported

- [x] Get bubblegum machine sprites from brainos

### CODE REVIEW

**Comment by LadyFowls** [comment]
!review

**Comment by alsoandanswer** [comment]
...why do we need an entirely new machine?

**Comment by LadyFowls** [comment]
> ...why do we need an entirely new machine?

Because reasons, one being its fundamentally different from vending machines

**Comment by BurgerLUA** [comment]
I thought this was a PR about an entirely different bubblegum.

**Comment by OneOneThreeEight** [comment]
same.

File: code/game/machinery/gumball.dm [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		update_icon()
+
+/obj/machinery/gumballmachine/update_icon()
+	switch(amountleft)
+		if(20)
+			icon_state = "[initialicon]_100"
+			desc = "A colorful [typeofcandy] machine. It is full!"
+		if(15)
+			icon_state = "[initialicon]_75"
+			desc = "A colorful [typeofcandy] machine. Some are missing"
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			desc = "A colorful [typeofcandy] machine. Some candy is missing."
```

File: code/game/machinery/gumball.dm [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		if(10)
+			icon_state = "[initialicon]_50"
+			desc = "A colorful [typeofcandy] machine. Its half full!"
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			desc = "A colorful [typeofcandy] machine. It's half full!"
```

File: code/game/machinery/gumball.dm [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		if(5)
+			icon_state = "[initialicon]_25"
+			desc = "A colorful [typeofcandy] machine. There is a few left"
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			desc = "A colorful [typeofcandy] machine. There is few candy left."
```

File: code/game/machinery/gumball.dm [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		if(0)
+			icon_state = "[initialicon]_empty"
+			desc = "A colorful [typeofcandy] machine. Its empty!"
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			desc = "A colorful [typeofcandy] machine. It's empty!"
```

File: code/game/machinery/gumball.dm [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	if (istype(W, /obj/item/weapon/spacecash))
+		var/obj/item/weapon/spacecash/C = W
+		if(!on)
+			to_chat(user, "<span class='warning'>There is no power!.</span>")
+			return
+		if(amountleft <= 0)
+			to_chat(user, "<span class='warning'>There is no more [typeofcandy] left!.</span>")
+			return
+		if(C.worth < gumprice)
+			to_chat(user, "<span class='warning'>You dont think this is enough to buy what you want from this.</span>")
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			to_chat(user, span("warning", "You don't think this is enough to buy what you want from this."))
```

File: code/game/machinery/gumball.dm [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+				spawn_money(changeleftover, src.loc, user)
+	if(istype(W, /obj/item/weapon/material/twohanded/baseballbat))
+		if(broken)
+			return
+		smashgumball()
+
+
+/obj/machinery/gumballmachine/proc/buygumball()
+	playsound(src.loc, 'sound/items/drumroll.ogg', 50, 0, -4)
+	sleep(50)
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]: Do we really want to use sleep here?

File: code/game/machinery/gumball.dm [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+	amountleft -= 1
+
+/obj/machinery/gumballmachine/proc/smashgumball()
+	icon_state = "[initialicon]_broken"
+	playsound(get_turf(src), 'sound/effects/Glassbr3.ogg', 75, 1)
+	if(amountleft)
+		var/amountleftinside = amountleft
+		for(var/i = 1;i<=amountleftinside,i++)
+			new vendingtype(src.loc)
+		src.visible_message("\The [src] shatters! [typeofcandy]'s falls out.", "You hear glass shatter!.")
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
		src.visible_message("\The [src] shatters! [typeofcandy]s falls out.", "You hear glass shatter!")
```

File: code/game/objects/items/weapons/chewables.dm [hunk 8]
```diff
@@ -0,0 +1,190 @@
+/obj/item/clothing/mask/chewable
+	name = "chewable item master"
+	desc = "if you are holding this, ahelp it."
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
	desc = "If you are seeing this, ahelp it."
```

File: code/game/objects/items/weapons/chewables.dm [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+			STOP_PROCESSING(SSprocessing, src)
+
+/obj/item/clothing/mask/chewable/process()
+	chew(1)
+	if(chewtime < 1)
+		spitout()
+
+/obj/item/clothing/mask/chewable/tobacco
+	name = "wad"
+	desc = "A chewy wad of tobacco. Cut in long strands and treated with syrups so it doesn't taste like a ash-tray when you stuff it into your face."
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
	desc = "A chewy wad of tobacco. Cut in long strands and treated with syrup so it doesn't taste like an ash-tray when you stuff it into your face."
```

File: code/game/objects/items/weapons/chewables.dm [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+	if (type_butt)
+		var/obj/item/butt = new type_butt(get_turf(src))
+		transfer_fingerprints_to(butt)
+		butt.color = color
+		if(brand)
+			butt.desc += " This one is \a [brand]."
+		if(ismob(loc))
+			var/mob/living/M = loc
+			if (!no_message)
+				to_chat(M, "<span class='notice'>You spit out the [name].</span>")
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
				to_chat(M, span("notice", "You spit out the [name]."))
```

File: code/game/objects/items/weapons/chewables.dm [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+		qdel(src)
+
+/obj/item/clothing/mask/chewable/tobacco/lenni
+	name = "chewing tobacco"
+	desc = "A chewy wad of tobacco. Cut in long strands and treated with syrups so it tastes less like a ash-tray when you stuff it into your face."
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
	desc = "A chewy wad of tobacco. Cut in long strands and treated with syrup so it tastes less like an ash-tray when you stuff it into your face."
```

File: code/game/objects/items/weapons/chewables.dm [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+	filling = list("nicotine" = 2)
+
+/obj/item/clothing/mask/chewable/tobacco/redlady
+	name = "chewing tobacco"
+	desc = "A chewy wad of fine tobacco. Cut in long strands and treated with syrups so it doesn't taste like a ash-tray when you stuff it into your face"
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
	desc = "A chewy wad of fine tobacco. Cut in long strands and treated with syrup so it doesn't taste like an ash-tray when you stuff it into your face."
```

File: code/game/objects/items/weapons/chewables.dm [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+	filling = list("nicotine" = 2)
+
+/obj/item/clothing/mask/chewable/tobacco/nico
+	name = "nicotine gum"
+	desc = "A chewy wad of synthetic rubber, laced with nicotine. Possibly the least disgusting method of nicotine delivery."
+	icon_state = "nic_gum"
+	type_butt = /obj/item/trash/spitgum
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
	type_butt = /obj/item/trash/spitgum

```

File: code/game/objects/items/weapons/chewables.dm [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+	desc = "A chewy wad of wadding material."
+	throw_speed = 0.5
+	icon_state = "chew"
+	type_butt = /obj/item/trash/spitgum
+	w_class = 1
+	slot_flags = SLOT_EARS | SLOT_MASK
+	chem_volume = 50
+	chewtime = 300
+	filling = list("sugar" = 2)
+
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]: Too many spaces left open.

File: code/game/objects/items/weapons/chewables.dm [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+
+	if (ismob(usr))
+		var/mob/M = usr
+		var/blowposition = M.loc
+		if (M.loc != blowposition)
+			to_chat(M, "<span class='notice'>You moved, popping your bubble!</span>")
+			spitout()
+		else
+			src.visible_message("[M.name] blows a bubble, what a cool cat!.", "You blow a bubble!.")
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
			src.visible_message("[M.name] blows a bubble, what a cool cat!", "You blow a bubble!")
```

File: code/game/objects/items/weapons/chewables.dm [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+	desc = "A chewy wad of fine synthetic rubber and artificial flavoring."
+	icon_state = "gum"
+	item_state = "gum"
+
+/obj/item/clothing/mask/chewable/candy/gum/New()
+	..()
+	reagents.add_reagent(pick("banana","berryjuice","grapejuice","lemonjuice","limejuice","orangejuice","watermelonjuice"),10)
+	color = reagents.get_color()
+
+/obj/item/clothing/mask/chewable/candy/gum/verb/blowbubble()
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]: You should add a cooldown to this, probably.

File: code/game/objects/items/weapons/chewables.dm [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+	type_butt = /obj/item/trash/lollibutt
+	icon_state = "lollipop"
+	item_state = "lollipop"
+
+/obj/item/clothing/mask/chewable/candy/lolli/New()
+	..()
+	reagents.add_reagent(pick("banana","berryjuice","grapejuice","lemonjuice","limejuice","orangejuice","watermelonjuice"),20)
+	color = reagents.get_color()
+
+/obj/item/clothing/mask/chewable/candy/lolli/meds
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]: This never gets medicine added to it.

File: code/game/objects/items/weapons/chewables.dm [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+		reagents.add_reagent(R, filling[R])
+
+/obj/item/clothing/mask/chewable/equipped(var/mob/living/user, var/slot)
+	..()
+	if(slot == slot_wear_mask)
+		var/mob/living/carbon/human/C = user
+		if(C.check_has_mouth())
+			START_PROCESSING(SSprocessing, src)
+		else
+			to_chat(user, "<span class='notice'>You don't have a mouth, and can't make much use of \the [src].</span>")
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
			to_chat(user, span("notice", "You don't have a mouth, and can't make much use of \the [src]."))
```

File: code/game/machinery/gumball.dm [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+			to_chat(user, "<span class='warning'>There is no power!.</span>")
+			return
+		if(amountleft <= 0)
+			to_chat(user, "<span class='warning'>There is no more [typeofcandy] left!.</span>")
+			return
+		if(C.worth < gumprice)
+			to_chat(user, "<span class='warning'>You dont think this is enough to buy what you want from this.</span>")
+			return
+		else
+			visible_message("<span class='info'>\The [usr] inserts a bill into \the [src].</span>")
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			visible_message(span("info", "\The [usr] inserts a bill into \the [src]."))
```

File: code/game/machinery/gumball.dm [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+		on = 0
+	else
+		on = 1
+
+
+/obj/machinery/gumballmachine/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if (istype(W, /obj/item/weapon/spacecash))
+		var/obj/item/weapon/spacecash/C = W
+		if(!on)
+			to_chat(user, "<span class='warning'>There is no power!.</span>")
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			to_chat(user, span("warning", "\The [src] has no power!"))
```

File: code/game/machinery/gumball.dm [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+			return
+		if(amountleft <= 0)
+			to_chat(user, "<span class='warning'>There is no more [typeofcandy] left!.</span>")
```
Comment (Geevies) [inline code/game/machinery/gumball.dm]:
```suggestion
			to_chat(user, span("warning", "There's no more [typeofcandy] left!"))
```

File: code/game/objects/items/weapons/chewables.dm [hunk 22]
```diff
@@ -0,0 +1,190 @@
...
+	set name = "Blow bubble"
+	set category = "Object"
+
+
+	if (ismob(usr))
+		var/mob/M = usr
+		var/blowposition = M.loc
+		if (M.loc != blowposition)
+			to_chat(M, "<span class='notice'>You moved, popping your bubble!</span>")
+			spitout()
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]: Why does moving and popping the bubble cause them to spit it out?

File: code/game/objects/items/weapons/chewables.dm [hunk 23]
```diff
@@ -0,0 +1,190 @@
...
+/obj/item/clothing/mask/chewable/candy/gum/verb/blowbubble()
+	set name = "Blow bubble"
+	set category = "Object"
+
+
+	if (ismob(usr))
+		var/mob/M = usr
+		var/blowposition = M.loc
+		if (M.loc != blowposition)
+			to_chat(M, "<span class='notice'>You moved, popping your bubble!</span>")
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
			to_chat(M, span("notice", "You moved, popping your bubble!"))
```

File: code/game/objects/items/weapons/chewables.dm [hunk 24]
```diff
@@ -0,0 +1,190 @@
...
+	filling = list("sugar" = 6)
+
+/obj/item/clothing/mask/chewable/candy/lolli/weak_meds/New()
+	..()
+	var/datum/reagent/payload = pick(list(
+				"dylovene",
+				"inaprovaline"))
+	reagents.add_reagent(payload, 15)
+	color = reagents.get_color()
+	desc = "[desc]. This one is labeled '[initial(payload.name)]'"
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
	desc = "[desc]. This one is labeled '[initial(payload.name)]'."
```

File: code/game/objects/items/weapons/chewables.dm [hunk 25]
```diff
@@ -0,0 +1,190 @@
...
+	if (ismob(usr))
+		var/mob/M = usr
+		var/blowposition = M.loc
+		if (M.loc != blowposition)
+			to_chat(M, "<span class='notice'>You moved, popping your bubble!</span>")
+			spitout()
+		else
+			src.visible_message("[M.name] blows a bubble, what a cool cat!.", "You blow a bubble!.")
+		
+
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]: Way too many spaces left open.

File: code/game/objects/items/weapons/chewables.dm [hunk 26]
```diff
@@ -0,0 +1,190 @@
...
+/obj/item/clothing/mask/chewable/dropped()
+	STOP_PROCESSING(SSprocessing, src)
+	..()
+
+obj/item/clothing/mask/chewable/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSprocessing, src)
+
+/obj/item/clothing/mask/chewable/proc/chew(amount)
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]: Why have an "amount" parameter if only one part is ever chewed at a time?

File: code/game/objects/items/weapons/chewables.dm [hunk 27]
```diff
@@ -0,0 +1,190 @@
...
+	if(reagents && reagents.total_volume)
+		if(ishuman(loc))
+			var/mob/living/carbon/human/C = loc
+			if (src == C.wear_mask && C.check_has_mouth())
+				reagents.trans_to_mob(C, REM, CHEM_INGEST, 0.2)
+		else
+			STOP_PROCESSING(SSprocessing, src)
+
+/obj/item/clothing/mask/chewable/process()
+	chew(1)
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm]: See above.

File: code/game/objects/items/weapons/chewables.dm (hunk 10 above)
Comment (NonQueueingMatt) [reply code/game/objects/items/weapons/chewables.dm]:
```suggestion
				to_chat(M, span("notice", "You spit out [name].")
```

**Comment by NonQueueingMatt** [comment]
Icon conflicts

File: code/game/machinery/gumball.dm [hunk 28]
```diff
@@ -0,0 +1,111 @@
+/obj/machinery/gumballmachine
+	name = "Gumball Machine"
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]:
```suggestion
	name = "gumball machine"
```
These aren't proper names.

File: code/game/machinery/gumball.dm [hunk 29]
```diff
@@ -0,0 +1,111 @@
...
+			desc = "A colorful [typeofcandy] machine. It's half full!"
+		if(5)
+			icon_state = "[initialicon]_25"
+			desc = "A colorful [typeofcandy] machine. There is few candy left."
+		if(0)
+			icon_state = "[initialicon]_empty"
+			desc = "A colorful [typeofcandy] machine. It's empty!"
+
+/obj/machinery/gumballmachine/proc/update_power()
+
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]:
```suggestion
```

File: code/game/machinery/gumball.dm [hunk 30]
```diff
@@ -0,0 +1,111 @@
...
+	amountleft -= 1
+
+/obj/machinery/gumballmachine/proc/smashgumball()
+	icon_state = "[initialicon]_broken"
+	playsound(get_turf(src), 'sound/effects/Glassbr3.ogg', 75, 1)
+	if(amountleft)
+		var/amountleftinside = amountleft
+		for(var/i = 1;i<=amountleftinside,i++)
+			new vendingtype(src.loc)
+		src.visible_message("\The [src] shatters! [typeofcandy]s falls out.", "You hear glass shatter!")
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]: Typeofcandy would be an improper type, so it'd be lowercase. Make it "src shatters and [typeofcandy] falls out."

File: code/game/machinery/gumball.dm [hunk 31]
```diff
@@ -0,0 +1,111 @@
...
+	stat |= BROKEN
+	anchored = 0
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]:
```suggestion
	anchored = FALSE
```

File: code/game/machinery/gumball.dm [hunk 32]
```diff
@@ -0,0 +1,111 @@
...
+	broken = 1
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]:
```suggestion
	broken = TRUE
```

File: code/game/machinery/gumball.dm [hunk 33]
```diff
@@ -0,0 +1,111 @@
...
+	amountleft = 0
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]:
```suggestion
	amountleft = FALSE
```

File: code/game/machinery/gumball.dm [hunk 34]
```diff
@@ -0,0 +1,111 @@
...
+
+/obj/machinery/gumballmachine/sucker
+	name = "Sucker Machine"
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]:
```suggestion
	name = "sucker machine"
```

File: code/game/objects/items/weapons/chewables.dm [hunk 35]
```diff
@@ -0,0 +1,191 @@
...
+	icon_state = "lollipop"
+
+/obj/item/clothing/mask/chewable/candy/lolli/meds/New()
+	..()
+	var/datum/reagent/payload = pick(list(
+				"paracetamol",
+				"tramadol",
+				"hyperzine",
+				"dylovene"))
+	reagents.add_reagent(payload, 15)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/chewables.dm]: Yeah, no, get rid of the hyperzine if it's gonna be 15 units.

File: code/game/objects/items/weapons/chewables.dm [hunk 36]
```diff
@@ -0,0 +1,191 @@
...
+	color = reagents.get_color()
+	desc = "[desc]. This one is labeled '[initial(payload.name)]'."
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/chewables.dm]:
```suggestion
	desc = "[desc] This one is labeled '[initial(payload.name)]'."
```
Desc already has a period.

File: code/game/machinery/gumball.dm [hunk 37]
```diff
@@ -0,0 +1,110 @@
...
+			desc = "A colorful [typeofcandy] machine. It is full!"
+		if(15)
+			icon_state = "[initialicon]_75"
+			desc = "A colorful [typeofcandy] machine. Some candy is missing."
+		if(10)
+			icon_state = "[initialicon]_50"
+			desc = "A colorful [typeofcandy] machine. It's half full!"
+		if(5)
+			icon_state = "[initialicon]_25"
+			desc = "A colorful [typeofcandy] machine. There is few candy left."
```
Comment (NonQueueingMatt) [inline code/game/machinery/gumball.dm]:
```suggestion
			desc = "A colorful [typeofcandy] machine. There are a few candies left."
```

File: code/game/machinery/gumball.dm [hunk 38]
```diff
@@ -0,0 +1,115 @@
...
+	else
+
+		update_power()
+
+		update_icon()
+
+
+/obj/machinery/gumballmachine/examine()
+	..()
+	to_chat(usr, span("notice", "\The [src] costs [gumprice] credits to use."))
```
Comment (skull132) [inline code/game/machinery/gumball.dm]: Use of `usr` in a non-interfacing proc.

Psure examine passes a user argument, use that.

File: code/game/machinery/gumball.dm [hunk 39]
```diff
@@ -0,0 +1,115 @@
...
+		if(amountleft <= 0)
+			to_chat(user, span("warning", "There's no more [typeofcandy] left!"))
+			return
+		if(C.worth < gumprice)
+			to_chat(user, span("warning", "You don't think this is enough to buy what you want from this."))
+			return
+		else
+			visible_message(span("info", "\The [usr] inserts a bill into \the [src]."))
+			var/changeleftover = C.worth - gumprice
+			usr.drop_from_inventory(C,get_turf(src))
```
Comment (skull132) [inline code/game/machinery/gumball.dm]:
```suggestion
			user.drop_from_inventory(C,get_turf(src))
```

File: code/game/machinery/gumball.dm [hunk 40]
```diff
@@ -0,0 +1,115 @@
...
+			to_chat(user, span("warning", "\The [src] has no power!"))
+			return
+		if(amountleft <= 0)
+			to_chat(user, span("warning", "There's no more [typeofcandy] left!"))
+			return
+		if(C.worth < gumprice)
+			to_chat(user, span("warning", "You don't think this is enough to buy what you want from this."))
+			return
+		else
+			visible_message(span("info", "\The [usr] inserts a bill into \the [src]."))
```
Comment (skull132) [inline code/game/machinery/gumball.dm]:
```suggestion
			visible_message(span("info", "\The [user] inserts a bill into \the [src]."))
```

File: code/game/objects/items/weapons/chewables.dm (hunk 13 above)
Comment (skull132) [reply code/game/objects/items/weapons/chewables.dm]: geeves wut

**Pull request merged**